<launch>
  <!-- Launch grap_test_rviz.launch to visualize this test -->

  <!-- Debug -->  
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix" value="" />
  <arg     if="$(arg debug)" name="launch_prefix" value="gdb --ex run --args" />

  <!-- Load the URDF, SRDF and other .yaml configuration files on the param server -->
  <arg name="jacob" default="false" />
  <arg name="baxter" default="false" />
  <arg name="hand" default="false" />

  <!-- JACOB ROBOT-->
  <group if="$(arg jacob)">
    <include file="$(find jacob_moveit_config)/launch/planning_context.launch">
      <arg name="load_robot_description" value="true"/>
    </include>

    <!-- Publish world frame -->
    <node pkg="tf" type="static_transform_publisher" name="tf_publisher_world" args="0 0 0 0 0 0 /world /base 25" />


    <!-- Which arm to use -->
    <arg name="ee_group_name" default="gripper"/>

    <!-- Start the test -->
    <node name="grasp_poses_visualizer" launch-prefix="$(arg launch_prefix)" pkg="moveit_grasps" 
	  type="grasp_poses_visualizer" output="screen" args="">
      <param name="ee_group_name" value="$(arg ee_group_name)"/>
      <rosparam command="load" file="$(find moveit_grasps)/config/jacob_grasp_data.yaml"/>
    </node>

  </group>

  <!-- BAXTER ROBOT-->
  <group if="$(arg baxter)">
    <include file="$(find baxter_moveit_config)/launch/planning_context.launch">
      <arg name="load_robot_description" value="true"/>
    </include>

    <!-- Publish world frame -->
    <node pkg="tf" type="static_transform_publisher" name="tf_publisher_world" args="0 0 0 0 0 0 /world /base 25" />

    <!-- Which arm to use -->
    <arg unless="$(arg hand)" name="ee_group_name" value="left_hand" />
    <arg     if="$(arg hand)" name="ee_group_name" value="right_hand" /> 

    <!-- Start the test -->
    <node name="grasp_poses_visualizer" launch-prefix="$(arg launch_prefix)" pkg="moveit_grasps" 
	  type="grasp_poses_visualizer" output="screen" args="">
      <param name="ee_group_name" value="$(arg ee_group_name)"/>
      <rosparam command="load" file="$(find moveit_grasps)/config/baxter_grasp_data.yaml"/>
    </node>
  </group>

</launch>
